cli/lsp/diagnostics.rs

```diff
@@ -514,6 +514,7 @@ impl DiagnosticsServer {
                         "Error generating TypeScript diagnostics: {}",
                         err
                       );
+                      token.cancel();
```

Comment:
The intent here is to avoid publishing the diagnostics if we errored (see https://github.com/denoland/deno/pull/23293/files#r1560000003). Not sure about this